cli/tests/pcluster/validators/test_cluster_validators.py

```diff
@@ -242,6 +242,31 @@ def test_efa_placement_group_validator(
             [{"IpProtocol": "-1", "UserIdGroupPairs": [{"UserId": "123456789012", "GroupId": "sg-12345678"}]}],
             None,
         ),
+        # right sg. Test when UserIdGroupPairs contains more entries
```

Comment:
Do we have an integration test with custom Security Group? If so I'd extend it to have more than 1 group pair.

Reply:
We do. But adding another group pair does not guarantee we could discover the issue from the integration tests. If the boto3 call returns the group pairs with the correct order, the test would now discover the issue. Moreover, it feels like a unit test to validate our validators are enough